### ISSUE TITLE
Possible bug - Causes Clickhouse to crash

### ISSUE THREAD

**Issue by peder1001** [ClickHouse/ClickHouse#61669]
**Describe what's wrong**
Clickhouse will sometimes crash when running the following query.
In our production environment we will get the dates from a MergeTree table.
In fiddle I have reproduced the scenario by inserting the dates into a table, which only sometimes causes an "internal error":
**{message: "internal error", code: 500}**

> A link to reproducer in:
https://fiddle.clickhouse.com/2cb6345e-6618-420a-9ef5-c8fef2fc1015

**Does it reproduce on the most recent release?**
Yes. Tested in 23.3.1.2823 and Clickhouse playground (fiddle) using latest.

**How to reproduce**
Check fiddle. You may have to run it a few times before you get the error.

**Expected behavior**
To not crash

**Error message and/or stacktrace**
From `SELECT * FROM system.crash_log ORDER BY event_time DESC LIMIT 100;`

`["2. ? @ 0x1297cdf6 in /usr/bin/clickhouse","3. ? @ 0x1297a5fb in /usr/bin/clickhouse","4. DB::ColumnLowCardinality::insertFrom(DB::IColumn const&, unsigned long) @ 0x13c36802 in /usr/bin/clickhouse","5. ? @ 0x14c3e9db in /usr/bin/clickhouse","6. ? @ 0x1394368f in /usr/bin/clickhouse","7. DB::ExecutionThreadContext::executeTask() @ 0x1494ce0a in /usr/bin/clickhouse","8. DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x14941e7b in /usr/bin/clickhouse","9. DB::PipelineExecutor::executeStep(std::atomic<bool>*) @ 0x14941268 in /usr/bin/clickhouse","10. DB::PullingPipelineExecutor::pull(DB::Chunk&) @ 0x14951e57 in /usr/bin/clickhouse","11. DB::PullingPipelineExecutor::pull(DB::Block&) @ 0x14952013 in /usr/bin/clickhouse","12. DB::MergeTask::ExecuteAndFinalizeHorizontalPart::executeImpl() @ 0x14407b2f in /usr/bin/clickhouse","13. DB::MergeTask::ExecuteAndFinalizeHorizontalPart::execute() @ 0x14407a8b in /usr/bin/clickhouse","14. DB::MergeTask::execute() @ 0x1440cafa in /usr/bin/clickhouse","15. DB::MergePlainMergeTreeTask::executeStep() @ 0x14797c31 in /usr/bin/clickhouse","16. DB::MergeTreeBackgroundExecutor<DB::DynamicRuntimeQueue>::routine(std::shared_ptr<DB::TaskRuntimeData>) @ 0x8985a4c in /usr/bin/clickhouse","17. DB::MergeTreeBackgroundExecutor<DB::DynamicRuntimeQueue>::threadFunction() @ 0x89856ca in /usr/bin/clickhouse","18. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false>, void*>) @ 0xe260ea5 in /usr/bin/clickhouse","19. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false>::ThreadFromGlobalPoolImpl<void ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::scheduleImpl<void>(std::function<void ()>, long, std::optional<unsigned long>, bool)::'lambda0'()>(void&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0xe263a15 in /usr/bin/clickhouse","20. ThreadPoolImpl<std::thread>::worker(std::__list_iterator<std::thread, void*>) @ 0xe25cc73 in /usr/bin/clickhouse","21. ? @ 0xe2628e1 in /usr/bin/clickhouse","22. ? @ 0x7f4a17a94ac3 in ?","23. ? @ 0x7f4a17b26850 in ?"]`

**Comment by alexey-milovidov** [comment]
1. It does not reproduce in the Fiddle.
2. The version 23.3.1 is not even the latest patch of the 23.3 LTS.
3. Your test does not contain either LowCardinality or MergeTree, as in the crash report.
4. You didn't mention whether you use the official build or not.
5. I will add a test anyways.

**Comment by alexey-milovidov** [comment]
I've added your example as a test, so it has run inside our CI with ASan, TSan, MSan, UBSan, with assertions, Stress test, Query Fuzzer, and for different platforms, including x86_64 and AArch64, and everything is just perfect.

I suspect that you are using not the official ClickHouse build, but some fake build.

**Comment by peder1001** [comment]
1. I get an 500 internal error message sometimes when running it in fiddle. I see now, that it could be unrelated to the SQL. 
2. This was the version I got when installing last summer using: `apt-get install -y clickhouse-server clickhouse-client`
3. No, you are right. On our server we are reading from a MergeTree table, but in Fiddle i used memroy.
4. `apt-get install -y clickhouse-server clickhouse-client` around August last year. As 

By the way, I updated clickhouse to 24.2.2.71 today and no longer experience the crash.
`$ sudo apt-get update
$ sudo apt-get install clickhouse-client clickhouse-server
$ sudo service clickhouse-server restart`
